torch/distributed/_tensor/experimental/attention.py

```diff
@@ -73,39 +83,70 @@ def _maybe_wait(tensor: torch.Tensor) -> torch.Tensor:
     return tensor
 
 
+def _partial_update(
+    original: torch.Tensor,
+    new: torch.Tensor,
+    dim: int,
+    n_chunks: int,
+    idx: int,
+    add: bool,
+) -> torch.Tensor:
+    chunks = list(original.chunk(n_chunks, dim=dim))
```

Comment:
worth adding a docstring for the behavior.  and should there be an assert for the relative size of new and original?